test_projects/cnode/test/example_test.exs

```diff
@@ -4,16 +4,86 @@ defmodule ExampleTest do
   test "init" do
     require Unifex.CNode
     assert {:ok, cnode} = Unifex.CNode.start_link(:example)
+
+    test_init(cnode)
+    test_unsigned_int(cnode)
+    test_string(cnode)
+    test_list(cnode)
+    test_list_as_string(cnode)
+    test_list_of_strings(cnode)
+    test_payload(cnode)
+    test_pid(cnode)
+    test_example_message(cnode)
+
+    test_undefined_function(cnode)
+    test_wrong_arguments(cnode)
```

Comment:
Some questions here. I am not sure if this is a good idea to test it all in a single unit test. I think we should make for each type separate unit test. On the other hand doing this in that way is a little faster because we create only one CNode and initialize its state only one time.

There are no all types tested here. Should I write it in this PR?

Should I write something similar for NIFs?

Reply:
> we create only one CNode and initialize its state only one time

exunit's setup_all should help

> Not all types are tested here. Should I write test for them in this PR?

Rather just create an issue with a list of those untested

> Should I write something similar for NIFs?

mention them in the issue

Reply:
Remaining tests should be implemented in #52